src/utils.jl

```diff
@@ -191,6 +191,34 @@ getstream(fmt, io::IOStream) = getstream(fmt, io, extract_filename(io))
 # assume OMETIFF if no format given
 getstream(io) = getstream(format"TIFF", io)
 
+@static if Sys.iswindows()
+    function _safe_open(f, filepath::String, mode="r", args...; kwargs...)
+        io = try
+            open(filepath, mode, args...; kwargs...)
+        catch err
+            # On Windows, trying to delete a file before garbage-collecting
+            # its corresponding mmapped-array results in an error.
+            # Here, this manifests as an error in precompiling the package,
+            # which is quite a serious problem.
+            # Thus try hard to make sure we free all the temporaries.
+            if err isa SystemError
+                @warn "failed to open file $filepath in \"$mode\" mode, retry after GC.gc()"
```

Comment:
I'm not sure either. Perhaps leave it a `@warn`, but perhaps we should give a likely cause (e.g., "this may be caused by overwriting a file previously opened with `mmap=true`").